common/tools/dev-tool/src/commands/run/testNodeTSInput.ts

```diff
@@ -2,24 +2,21 @@
 // Licensed under the MIT license
 
 import { leafCommand, makeCommandInfo } from "../../framework/command";
-import { runTestsWithProxyTool } from "../../util/testUtils";
+import { concatArguments, runTestsWithProxyTool } from "../../util/testUtils";
 
 export const commandInfo = makeCommandInfo(
   "test:node-ts-input",
-  "runs the node tests using mocha with the default and the provided options; starts the proxy-tool in record and playback modes",
-  {
-    mocha: {
-      kind: "string",
-      description:
-        "Mocha options along with the test files(glob pattern) in TS as expected by mocha",
-      default: '--timeout 1200000 --exclude "test/**/browser/*.spec.ts" "test/**/*.spec.ts"'
-    }
-  }
+  "runs the node tests using mocha with the default and the provided options; starts the proxy-tool in record and playback modes"
 );
 
 export default leafCommand(commandInfo, async (options) => {
+  const defaultMochaArgs =
+    "-r esm -r ts-node/register --reporter ../../../common/tools/mocha-multi-reporter.js --full-trace";
+  const mochaArgs = options["--"]?.length
+    ? concatArguments(options["--"])
+    : '--timeout 1200000 --exclude "test/**/browser/*.spec.ts" "test/**/*.spec.ts"';
```

Comment:
Nit: this pattern is used a few times. Is it merely joining with spaces? What about:

```typescript
options["--"]?.join(" ") ?? <default>
```

Reply:
Ah, yesss!

Reply:
> options["--"]?.join(" ") ?? <default>

Just this isn';t enough - for [] empty arrays.
going with this `options["--"]?.length ? options["--"]?.join(" ") : "--single-run"` instead

Reply:
Ah I see, oh well!